### PR TITLE
feat: added auth and private routes

### PR DESCRIPTION
**Pull request by srishti-maurya**
## Added 
- authentication (login, logout and signup)
- private routes

### CODE REVIEW

**Review by KaustubhM1997 — APPROVED** [review]
The UI looks great!

File: src/frontend/components/Nav.jsx [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
+              </button>
+            </>
+          ) : (
+            <button
+              className="btn btn-sm primary"
+              onClick={() => navigate("/login")}
+            >
+              Login
+            </button>
+          )}
```
Comment (pritam-kr) [inline src/frontend/components/Nav.jsx:31]: I think it can be done with a single button only.

File: src/frontend/context/auth-context.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  const loginHandler = async (e) => {
+    e.preventDefault();
+    try {
+      const { data } = await loginRequest(loginInput);
+      localStorage.setItem("token", JSON.stringify(data.encodedToken));
+      localStorage.setItem("isLoggedIn", JSON.stringify(true));
+      setToken(data.encodedToken);
+      setIsLoggedIn(true);
+      //   setTimeout(() => {
```
Comment (pritam-kr) [inline src/frontend/context/auth-context.js:30]: Remove these things if not in use

**Comment by pritam-kr** [comment]
Everything is looking good for me. 
- clean code and UI as per user perspective

**Review by pritam-kr — APPROVED** [review]

**Pull request merged**